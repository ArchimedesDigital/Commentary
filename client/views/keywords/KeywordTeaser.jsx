import Card from 'material-ui/Card';
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

KeywordTeaser = React.createClass({

  getChildContext() {
    return { muiTheme: getMuiTheme(baseTheme) };
  },

  childContextTypes: {
    muiTheme: React.PropTypes.object.isRequired
  },

  propTypes: {
    keyword: React.PropTypes.object.isRequired
  },


  render() {
    let keyword = this.props.keyword;
    let keyword_url = "/keywords/" + keyword.slug ;

     return (
       <div  class="keyword-teaser wow fadeInUp" data-wow-duration="0.2s">
        <a class="keyword-title" href="/keyword/show/{keyword.id}" >
            {keyword.title}
        </a>
        <span class="keyword-comment-count">437 Comments</span>
        <span class="keyword-description">{keyword.description}</span>

      </div>

      );
    }

});
