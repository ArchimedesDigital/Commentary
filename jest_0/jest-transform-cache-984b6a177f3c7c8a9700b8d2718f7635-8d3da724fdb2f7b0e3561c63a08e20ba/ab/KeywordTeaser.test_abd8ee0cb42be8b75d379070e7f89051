/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_KeywordTeaser = require('./KeywordTeaser'); /* istanbul ignore next */var _KeywordTeaser2 = _interopRequireDefault(_KeywordTeaser);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('KeywordTeaser', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_KeywordTeaser2.default, /* istanbul ignore next */{
			keyword: {
				title: 'Test',
				slug: 'test',
				description: 'Test description' } })).



		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: