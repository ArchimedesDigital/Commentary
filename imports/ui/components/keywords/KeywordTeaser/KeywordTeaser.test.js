import React from 'react';
import renderer from 'react-test-renderer';

// component:
import KeywordTeaser from './KeywordTeaser';

describe('KeywordTeaser', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<KeywordTeaser />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
