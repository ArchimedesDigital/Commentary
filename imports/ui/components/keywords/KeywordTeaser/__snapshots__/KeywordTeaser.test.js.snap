// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordTeaser renders correctly 1`] = `
<div
  className="keyword-teaser clearfix wow fadeInUp "
  data-wow-duration="0.2s"
  onClick={[Function]}
>
  <h4
    className="keyword-title"
  >
    Test
  </h4>
  <i
    className="mdi mdi-plus"
  />
  <i
    className="mdi mdi-minus"
  />
  <span
    className="keyword-comment-count"
  >
    (
     Comments)
  </span>
  <hr />
  <span
    className="keyword-description"
  >
     
    Test description
     
  </span>
  <a
    className="keyword-read-more"
    href="/tags/test"
  >
    Read more
  </a>
</div>
`;
